pandas/core/dtypes/cast.py

```diff
@@ -1450,8 +1450,16 @@ def find_common_type(types):
     if all(is_dtype_equal(first, t) for t in types[1:]):
         return first
 
+    # get unique types (dict.fromkeys is used as order-preserving set())
+    types = list(dict.fromkeys(types).keys())
+
     if any(isinstance(t, ExtensionDtype) for t in types):
-        return np.object
+        for t in types:
+            if isinstance(t, ExtensionDtype):
```

Comment:
this seems a bit arbitrary. we try the types in order to see if they are a unifying type? do we actually test this?

Reply:
Yes, so it is not arbitrary, but in order. 
(eg the order of the arguments passed to concat. This order for example also determines the class used for the return value)

I don't think we have internally dtypes where this order actually matters, right now (meaning, where a different order would give a different result). So we don't *really* test the order-dependency.

Reply:
Tom and I also had some discussion about this in https://github.com/pandas-dev/pandas/issues/22994#issuecomment-614203960